fw/http_frame.c

```diff
@@ -1848,18 +1855,19 @@ tfw_h2_frame_process(TfwConn *c, struct sk_buff *skb, struct sk_buff **next)
 		h2->data_off = 0;
 		/* The skb will not be parsed, just flags will be checked. */
 		r = tfw_http_msg_process_generic(c, h2->cur_stream, pskb, next);
-
-		/* TODO #1490: Check this place, when working on the task. */
-		if (r && r != T_DROP) {
-			WARN_ON_ONCE(r == T_POSTPONE);
-			kfree_skb(nskb);
-			goto out;
-		}
+		/*
+		 * If Conn_Stop bit is not set we don't send error response
+		 * to client, so we don't need to process WINDOW_UPDATE frames
+		 * and can immediately close connection.
+		 */ 
+		TFW_CONN_PROCESS_RESULT(r, was_stopped,
+					!(TFW_CONN_TYPE(c) & Conn_Stop), nskb);
```

Comment:
The `tfw_http_msg_process_generic()` and `TFW_CONN_PROCESS_RESULT()` are identical, so instead of the macro we can just introduce a function to call  `tfw_http_msg_process_generic()` and process the result